src/main/java/com/codingParkFun/rmsbackend/controller/SemesterController.java

```diff
@@ -0,0 +1,64 @@
+package com.codingParkFun.rmsbackend.controller;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.codingParkFun.rmsbackend.exception.IdDoesNotExistException;
+import com.codingParkFun.rmsbackend.exception.SchoolIdMismatchException;
+import com.codingParkFun.rmsbackend.model.Semester;
+import com.codingParkFun.rmsbackend.service.SemesterService;
+
+@RestController
+@RequestMapping("/schools/{schoolId}/semesters")
+public class SemesterController {
+
+	@Autowired
+	SemesterService semesterService;
+
+	@PostMapping
+	@ResponseStatus(HttpStatus.CREATED)
+	public void createSemester(@PathVariable long schoolId, @RequestBody Semester semester) {
+		if (semester.getSchoolId() == null) {
+			semester.setSchoolId(schoolId);
+		}
+
+		if (semester.getSchoolId() != schoolId) {
+			throw new SchoolIdMismatchException("School Id mismatch");
+		}
+
+		semesterService.create(semester);
+	}
```

Comment:
I think the check we are doing here is not correct. we need to create semester inside a schoolId. after creation it stores the schoolId inside semester. but here we are already having schoolid inside semester i didn't understand y like this?

Reply:
Semester creation REST API expects an object with all the properties except id from the client. That object can consist of school Id as well. Now I added a check to validate if user have provided the same schoolId as path param.